src/algorithms/gaco.cpp

```diff
@@ -270,18 +274,56 @@ population gaco::evolve(population pop) const
         // 5 - use pheromone values to generate new ants (i.e., individuals)
         generate_new_ants(popold, dist, gauss, prob_cumulative, sigma, new_ants, sol_archive);
 
-        for (population::size_type i = 0; i < pop_size; ++i) {
-            // I compute the fitness for each new individual which was generated in the generated_new_ants(..)
-            // function
-            for (decltype(new_ants[i].size()) ii = 0u; ii < new_ants[i].size(); ++ii) {
-                ant[ii] = new_ants[i][ii];
+        if (m_bfe) {
+            // bfe is available:
+            unsigned pos = 0u;
```

Comment:
Since this ``pos`` variable is used to index into the ``ants`` vector, its type should be ``decltype(ants.size())``, not ``unsigned``. Same with similar index variables below.